src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * duke.Storage class allows to save and load data on the hard disk on the file saved-tasks.txt
+ * under directory ./data.
+ */
+public class Storage {
+
+    public String path;
+
+    public Storage(String path) {
+        this.path = path;
+    }
+
+
+    /**
+     * Stores the current task List on the hard disk.
+     */
+    public void saveData() {
+        try {
+            File file = new File("./data/saved-tasks.txt");
+            if(!file.exists()) {
+                file.createNewFile();
+            }
+            FileWriter writer = new FileWriter(file, false);
+            for(Task task: Tasklist.dukeList) {
+                String commandLine = "";
+                if(task instanceof Deadline) {
+                    commandLine = "D | " + (task.getIsDone() ? "1 | " : "0 | ") + task.getDescription() + " | " + ((Deadline) task).getTime() + '\n';
+                } else if(task instanceof Event) {
+                    commandLine = "E | " + (task.getIsDone() ? "1 | " : "0 | ") + task.getDescription() + " | " + ((Event) task).getTime() + '\n';
+                } else {
+                    commandLine =  "T | " + (task.getIsDone() ? "1 | " : "0 | ") + task.getDescription() + '\n';
```

Comment:
I think this chunk is not very reader-friendly 😬? You could try splitting them up into a few lines. Remember to follow the convention (e.g the number of tabs, etc)